app/models/user.rb

```diff
@@ -1,2 +1,14 @@
 class User < ApplicationRecord
+  has_many :posts, class_name: 'Post'
+  has_many :likes, class_name: 'Like'
+  has_many :comments, class_name: 'Comment'
+
+  def recent_posts
+    Post.where(user_id: id).order(created_at: :desc).limit(3)
+  end
+
+  def update_posts_counter
+    posts_counter = Post.where(user_id: id).count
+    update(postsCounter: posts_counter)
+  end
```

Comment:
- [ ] The `update_posts_counter` should be inside the `post.rb` model. So it will be nice to refactor your code and fix this.

**Optional**

- [ ] It would be nice to use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the #update_post_counter method every time a new post has been saved to database, then make the update_post_counter method private like so:

```ruby
# app/models/post.rb
  after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end
```

Reply:
what is `author` in this statement?

Reply:
@Amrhub, Yes, you're correct. But you didn't use author as I can see you used the User class directly. You were supposed to reference the author_id in the User class in the previous project but you referenced the user_id. So yours might look something like `user.increment!....`.

Reply:
I did that @julie-ify 
Thanks for clarifying, that was my reason beyond asking but I tested and then when it worked I figured I was correct 😆